Dockerfile

```diff
@@ -1,43 +1,45 @@
-    && apt-get -y update && apt-get -y upgrade \
-    && apt-get -y --no-install-recommends install \
-        cron \
-        logrotate \
-        monitoring-plugins \
-        supervisor \
-        nginx \
-        wallarm-node \
-        libnginx-mod-http-wallarm \
-        collectd-utils \
-        curl \
-        iptables \
-        bsdmainutils \
-        sudo \
-        socat \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
-    && chown -R wallarm:wallarm /var/lib/wallarm-tarantool \
-    && sed -i 's|session    required     pam_loginuid.so|session    optional     pam_loginuid.so|g' /etc/pam.d/cron
-
-RUN cp /usr/share/doc/libnginx-mod-http-wallarm/examples/wallarm-status.conf /etc/nginx/conf.d/
+ARG WLRM_FOLDER
+ARG TARGETPLATFORM
+
+RUN apk update && apk upgrade && apk add --no-cache bash runuser socat logrotate libgcc && rm -r /var/cache/apk/* && \
+  addgroup -S wallarm && \
+  adduser -S -D -G wallarm -h /opt/wallarm wallarm
+
+COPY --chown=wallarm:wallarm build/$TARGETPLATFORM/opt /opt
+COPY build/$TARGETPLATFORM/opt/wallarm/modules/${WLRM_FOLDER} /usr/lib/nginx/modules
+
+COPY build-scripts/check_sig.sh /opt/wallarm/check_sig.sh
+RUN /bin/sh -c '/opt/wallarm/check_sig.sh' && rm /opt/wallarm/check_sig.sh
+
+RUN sed -i \
+    -e '/WALLARM_COMPONENT_NAME/d' \
+    -e '/WALLARM_COMPONENT_VERSION/d' \
+    -e '/SLAB_ALLOC_ARENA/d' \
+    /opt/wallarm/env.list && \
+    sed -i \
+    -e '/FQDNLookup/s/no/true/' \
+    -e '/DeleteSocket/aSocketGroup "wallarm"' \
+    /opt/wallarm/etc/collectd/wallarm-collectd.conf
+
 COPY scripts/init /usr/local/bin/
 COPY scripts/registernode_loop /usr/local/bin/
 
-COPY conf/supervisord.conf /etc/supervisor/
-COPY conf/supervisord.filtering.conf /etc/supervisor/supervisord.filtering.conf.example
-COPY conf/supervisord.post-analytics.conf /etc/supervisor/supervisord.post-analytics.conf.example
+# COPY conf/supervisord.conf /etc/supervisor/
+# COPY conf/supervisord.filtering.conf /etc/supervisor/supervisord.filtering.conf.example
+# COPY conf/supervisord.post-analytics.conf /etc/supervisor/supervisord.post-analytics.conf.example
+
+COPY conf/wallarm-status.conf /etc/nginx/conf.d/
+COPY conf/default /etc/nginx/conf.d/default.conf
+COPY conf/proxy_params /etc/nginx/
 
-COPY conf/logrotate.conf /etc/
-COPY conf/default /etc/nginx/sites-enabled/
-COPY conf/collectd.conf /etc/collectd/
+# TODO: cant use old collectd.conf because a lot of plugins that load link dynamically to libs that we dont have installed anymore
+# COPY conf/collectd.conf /opt/wallarm/etc/collectd/wallarm-collectd.conf
 
```

Comment:
```suggestion
USER wallarm
```

Reply:
Couldn't do this because init script required permissions broader than that
IIRC most of the time we drop permissions anyway

Reply:
@qqshka we to do something with that

Reply:
It's always been like that with docker container
Init script created symlinks, edited configs in /etc/nginx etc

Hypothetically we may try to work around this if necessary

Reply:
Using root user is an anti-pattern, in Wallarm Sidecar we use a non-root user

Reply:
Changing this turned out to be easier than thought, fixed